go/v1beta1/storage/esutil/index.go

```diff
@@ -0,0 +1,235 @@
+
+	IndexPrefix = "grafeas"
+	AliasPrefix = "grafeas"
+)
+
+type IndexManager interface {
+	LoadMappings(mappingsDir string) error
+	CreateIndex(ctx context.Context, info *IndexInfo, checkExists bool) error
+
+	ProjectsIndex() string
+	ProjectsAlias() string
+
+	OccurrencesIndex(projectId string) string
+	OccurrencesAlias(projectId string) string
+
+	NotesIndex(projectId string) string
+	NotesAlias(projectId string) string
+}
+
+type EsIndexManager struct {
+	logger            *zap.Logger
+	client            *elasticsearch.Client
+	projectMapping    *VersionedMapping
+	occurrenceMapping *VersionedMapping
+	noteMapping       *VersionedMapping
+}
+
+type VersionedMapping struct {
+	Version  string                 `json:"version"`
+	Mappings map[string]interface{} `json:"mappings"`
+}
+
+type IndexInfo struct {
+	Index        string
+	Alias        string
+	DocumentKind string
+}
+
+type IndexNameParts struct {
+	DocumentKind string
+	Version      string
+	ProjectId    string
+}
+
+func NewEsIndexManager(logger *zap.Logger, client *elasticsearch.Client) *EsIndexManager {
+	return &EsIndexManager{
+		client: client,
+		logger: logger,
+	}
+}
+
+var (
+	ioutilReadDir  = ioutil.ReadDir
+	ioutilReadFile = ioutil.ReadFile
+)
+
+func (em *EsIndexManager) LoadMappings(mappingsDir string) error {
+	files, err := ioutilReadDir(mappingsDir)
+	if err != nil {
+		return err
+	}
+	currentDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	for _, file := range files {
+		if file.IsDir() {
+			continue
+		}
+
+		documentKind := strings.TrimSuffix(file.Name(), filepath.Ext(file.Name()))
+		filePath := path.Join(currentDir, mappingsDir, file.Name())
+		versionedMappingJson, err := ioutilReadFile(filePath)
+
+		if err != nil {
+			return err
+		}
+		var mapping VersionedMapping
+
+		if err := json.Unmarshal(versionedMappingJson, &mapping); err != nil {
+			return err
+		}
+
+		switch documentKind {
+		case ProjectDocumentKind:
+			em.projectMapping = &mapping
+		case OccurrenceDocumentKind:
+			em.occurrenceMapping = &mapping
+		case NoteDocumentKind:
+			em.noteMapping = &mapping
+		default:
+			return fmt.Errorf("unrecognized document kind mapping: %s", documentKind)
+		}
+	}
+
+	return nil
+}
+
+func (em *EsIndexManager) CreateIndex(ctx context.Context, info *IndexInfo, checkExists bool) error {
+	log := em.logger.Named("CreateIndex").With(zap.String("index", info.Index))
+
+	if checkExists {
+		res, err := em.client.Indices.Exists([]string{info.Index}, em.client.Indices.Exists.WithContext(ctx))
+		if err != nil {
+			return fmt.Errorf("error checking if index %s exists: %s", info.Index, err)
+		}
+
+		if res.StatusCode != http.StatusOK && res.StatusCode != http.StatusNotFound {
+			log.Error("error checking if index exists", zap.String("response", res.String()), zap.Int("status", res.StatusCode))
+
+			return fmt.Errorf("unexpected status code (%d) when checking if index exists", res.StatusCode)
+		}
+
+		if !res.IsError() {
```

Comment:
maybe it would be useful to leave a comment here indicating that we expect an error to occur here if the index already exists.  it took me a bit to figure out what the code was trying to do if `checkExists` is true.